### PR TITLE
:sparkles: [#2088] Implement Zaken search for eHerkenning users

### PR DESCRIPTION
**Pull request by stevenbal**
issue: https://taiga.maykinmedia.nl/project/open-inwoner/issue/2088

### CODE REVIEW

File: src/open_inwoner/openzaak/clients.py [hunk 1]
```diff
@@ -40,11 +40,32 @@
...
 class ZakenClient(APIClient):
+    def fetch_cases(
+        self,
+        user_bsn=None,
+        user_kvk_or_rsin=None,
+        max_requests=4,
+        identificatie=None,
+        vestigingsnummer=None,
+    ):
+        if user_bsn:
```
Comment (Bartvaderkin) [inline src/open_inwoner/openzaak/clients.py:51]: For correctness we could check if not both `user_bsn` and `user_kvk_or_rsin` are supplied as they're mutually exclusive. Or `vestigingsnummer` and `user_bsn` etc.

Also some typing in the signature would be good.

File: src/open_inwoner/openzaak/utils.py [hunk 2]
```diff
@@ -143,3 +144,25 @@ def translate_single_status(status_text: str) -> str:
         )
     except StatusTranslation.DoesNotExist:
         return ""
+
+
+def get_fetch_parameters(request) -> dict:
```
Comment (Bartvaderkin) [inline src/open_inwoner/openzaak/utils.py:149]: Good idea to clean this up. Maybe use a name like `get_user_fetch_parameters()` so we keep it for user-related params only?

File: src/open_inwoner/search/views.py [hunk 3]
```diff
@@ -62,10 +64,9 @@ def search(self, form):
             self.log_user_action(user, _("search query: {query}").format(query=query))
 
         # Check if the query exactly matches with a case that belongs to the user
-        # TODO should be implemented for KVK as well
-        if hasattr(self.request.user, "bsn"):
+        if search_params := get_fetch_parameters(self.request):
```
Comment (Bartvaderkin) [inline src/open_inwoner/search/views.py:67]: Here you see it's unclear: this makes get_fetch_parameters() look like magic. We're getting the fetch params for the user, not a generic mystery search.

**Review by Bartvaderkin — CHANGES_REQUESTED** [review]
Some small notes.

**Review by Bartvaderkin — CHANGES_REQUESTED** [review]

File: src/open_inwoner/openzaak/clients.py [hunk 4]
```diff
@@ -42,12 +42,16 @@
...
-        vestigingsnummer=None,
+        user_bsn: Optional[str] = None,
+        user_kvk_or_rsin: Optional[str] = None,
+        max_requests: Optional[int] = 4,
+        identificatie: Optional[str] = None,
+        vestigingsnummer: Optional[str] = None,
     ):
+        assert not (
+            user_bsn and (user_kvk_or_rsin or vestigingsnummer)
+        ), "either `user_bsn` or `user_kvk_or_rsin`/`vestigingsnummer` should be supplied, not both"
```
[the remark this answers is not in the document]
Comment (stevenbal) [reply src/open_inwoner/openzaak/clients.py:53]: Oh TIL, I always assumed `Optional` was for optional arguments, but you're right

**Review by Bartvaderkin — APPROVED** [review]

**Pull request merged**